buildlib/azure-pipelines-perf.yml

```diff
@@ -128,5 +130,11 @@ stages:
             clean: true
           - bash: |
               set -x
+              echo 'Retain 3 latest workdirs for manual repro; clean the rest'
+              wrk_dirs="/hpc/scrap/azure/$(Build.DefinitionName)/"
+              for dir in $(ls -lt "$wrk_dirs" | tail -n +4 | awk '{print $NF}'); do
```

Comment:
Potentially unsafe - please check shellcheck warnings

Reply:
We can safely disregard shellcheck[SC2012](https://www.shellcheck.net/wiki/SC2012), as we do not generate non-alphanumeric filenames.